tm2py/components/network/transit/transit_assign.py

```diff
@@ -1,9 +1,632 @@
+    if mode_char in ["l", "x"]:
+        eawt_factor = 0.4
+    elif mode_char in ["h", "c", "f"]:
+        eawt_factor = 0.2
+    else:
+        eawt_factor = 1
+    return eawt * eawt_factor
+
+
+def calc_adj_headway(transit_volume, transit_boardings, headway, capacity, segment):
+    prev_hdwy = segment["@phdwy"]
+    delta_cap = max(capacity - transit_volume + transit_boardings, 0)
+    adj_hdwy = min(max_hdwy, prev_hdwy * min((transit_boardings+1) / (delta_cap+1), 1.5))
+    adj_hdwy = max(headway, adj_hdwy)
+    return adj_hdwy
+
+def calc_headway(transit_volume, transit_boardings, headway, capacity, segment):
+    vcr = transit_volume / capacity
+    eawt = calc_eawt(segment, vcr, segment.line.headway)
+    adj_hdwy = calc_adj_headway(transit_volume, transit_boardings, headway, capacity, segment)
+    return adj_hdwy + eawt
+"""
+
+EmmeTransitJourneyLevelSpec = List[
+    Dict[
+        str,
+        Union[
+            str, bool, List[Dict[str, Union[int, str]]], Dict[str, Union[float, str]]
+        ],
+    ]
+]
+EmmeTransitSpec = Dict[
+    str,
+    Union[
+        str,
+        Dict[str, Union[str, float, bool, Dict[str, Union[str, float]]]],
+        List[str],
+        EmmeTransitJourneyLevelSpec,
+        None,
+    ],
+]
 
 
 class TransitAssignment(Component):
     """Run transit assignment."""
+
+    def __init__(self, controller: "RunController"):
+        """Constructor for TransitAssignment.
+
+        Args:
+            controller: RunController object.
+        """
+        super().__init__(controller)
+        self.config = self.controller.config.transit
+        self.demand = PrepareTransitDemand(self.controller)
+        self._num_processors = tools.parse_num_processors(
+            self.config.emme.num_processors
+        )
+        self._time_period = None
+        self._scenario = None
+
+    @LogStartEnd("Transit assignments")
+    def run(self):
+        """Run transit assignments."""
+        emmebank_path = self.get_abs_path(
+            self.controller.config.emme.transit_database_path
+        )
+        emmebank = self.controller.emme_manager.emmebank(emmebank_path)
+        use_ccr = False
+        if self.controller.iteration >= 1:
+            use_ccr = self.config.transit.use_ccr
+            self.demand.run()
+        else:
+            self.demand.create_zero_matrix(emmebank)
+        for self._time_period in self.time_period_names:
+            msg = f"Transit assignment for period {self._time_period}"
+            with self.logger.log_start_end(msg):
+                self._scenario = self.get_emme_scenario(emmebank, self._time_period)
+                if use_ccr:
+                    self._run_ccr_assign()
+                    self._calc_segment_ccr_penalties()
+                else:
+                    self._run_extended_assign()
+                if self.config.transit.output_stop_usage_path is not None:
+                    network, class_stop_attrs = self._calc_connector_flows()
```

Comment:
do we want to calculate the connector flows to the network even if we don't export them?